config/routes.rb

```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
   get 'welcome/index'
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :teams do
+    resources :users
```

Comment:
Similar to the scope, this is a good place to specify which controller actions are needed. Obvz you need the basics index, show, edit, update but is there a need for a create/destroy route in this GUI here? Realistically you would have functionality later that connects to workday as the source of truth and does the create/destroy(move to inactive) here. 

Not to de"rail" too much (#womp), but check out the [paranoid gem](https://github.com/rubysherpas/paranoia) for your user model. tl;dr it adds a `deleted_at` timestamp column to your table and intercepts destroy commands.

Reply:
Good call - at least for the purpose of presenting we plan to leave the create action since we are not sure if we will be able to tie in the functionality of pulling users from Workday and still want to be able to add new users.

I think destroy wont be needed here, and we could definitely use something like the paranoid gem that adds a "deleted_at" timestamp when the user is removed. We already added an "active" column that was supposed to cover if a user has been "deleted", so will need to decide if it would be better to add the gem or try to do it ourselves with the "active" column.

I'll try to swing by office hours when I get the chance to go over this ^^^ as well as some questions re: setting up the Users scope in the model vs. the controller - seems better to set it in the model but I am less familiar with how to set that up/how it functions in the rest of the app.

Reply:
Oh right the active column. That should be sufficient then at least for now. The only benefit to a timestamp column is that you knw when something happened, but probably not critical to this. 

Putting scopes in the model is generally a better decision, especially if you need several. It's really just about making the code as readable as possible through your scope names. Your use case is fairly easy though:

```ruby
# in the controller

# controller
def index
  @users = User.where(active: true)
end


# in the model

# model
scope :active, -> { where(active: true) }
# controller
def index
  @users = User.active
end

Reply:
And remember, you can chain multiple scopes, if you require a certain order to them or something.